sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -721,16 +791,37 @@ private[execution] final class SlidingWindowFunctionFrame(
  */
 private[execution] final class UnboundedWindowFunctionFrame(
     target: MutableRow,
-    processor: AggregateProcessor) extends WindowFunctionFrame {
+    processor: AggregateProcessor,
+    excludeSpec: ExcludeClause) extends WindowFunctionFrame {
+
+  // Refer to the comment below in function prepare()
+  private[this] val buffer = new util.ArrayList[InternalRow]()
 
   /** Prepare the frame for calculating a new partition. Process all rows eagerly. */
   override def prepare(rows: RowBuffer): Unit = {
     val size = rows.size()
     processor.initialize(size)
-    var i = 0
-    while (i < size) {
-      processor.update(rows.next())
-      i += 1
+    if (excludeSpec.excludeType != ExcludeNoOthers) {
+      // For the exclude cases, the content of the window frame is always changing
+      // along with the current row. For example, a row that was not included before
+      // needs to be included now, or a row that was included before needs to be excluded now.
+      // So intermediate buffer is used for now..
+      // TODO: For potential performance gains, we need to look into the AggregateProcessor
```

Comment:
There is no way this is possible. Think of a MIN aggregate for instance. What you can do is merge multiple buffers.